presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergDistributedSmokeTestBase.java

```diff
@@ -1233,4 +1242,80 @@ public void testUnregisterTable()
         assertUpdate("CALL system.unregister_table('" + schemaName + "', '" + tableName + "')");
         assertFalse(getQueryRunner().tableExists(getSession(), tableName));
     }
+
+    @Test
+    public void testDeleteUnPartitionedTable()
+    {
+        Session session = getSession();
+
+        // Test with default delete_mode i.e. merge-on-read
+        String tableNameMor = "test_delete_mor";
+
+        assertUpdate("CREATE TABLE " + tableNameMor + " (id integer, value integer) WITH (format_version = '2')");
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (1, 10)", 1);
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (2, 13)", 1);
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (3, 5)", 1);
+        assertQuery("SELECT * FROM " + tableNameMor, "VALUES (1, 10), (2, 13), (3, 5)");
+        assertUpdate("DELETE FROM " + tableNameMor + " WHERE id = 1", 1);
+        assertQuery("SELECT * FROM " + tableNameMor, "VALUES (2, 13), (3, 5)");
+
+        dropTable(session, tableNameMor);
+
+        // Test with delete_mode set to copy-on-write
+        String tableNameCor = "test_delete_cor";
+        @Language("RegExp") String errorMessage = "This connector only supports delete where one or more partitions are deleted entirely";
+
+        assertUpdate("CREATE TABLE " + tableNameCor + " (id integer, value integer) WITH (format_version = '2', delete_mode = 'copy-on-write')");
+        assertUpdate("INSERT INTO " + tableNameCor + " VALUES (1, 5)", 1);
+        assertQuery("SELECT * FROM " + tableNameCor, "VALUES (1, 5)");
+        assertQueryFails("DELETE FROM " + tableNameCor + " WHERE value = 5", errorMessage);
+
+        dropTable(session, tableNameCor);
+    }
+
```

Comment:
```suggestion
@Test
    public void testDeleteOnTableWithPartitionEvolution()
    {
        String tableName = "test_delete_on_table_with_partition_evolution";
        Session session = getSession();

        assertUpdate("CREATE TABLE " + tableName + " (a integer, b varchar) WITH (format_version = '2')");
        assertUpdate("INSERT INTO " + tableName + " VALUES (1, '1001'), (2, '1002'), (3, '1003')", 3);
        assertQuery("SELECT * FROM " + tableName, "VALUES (1, '1001'), (2, '1002'), (3, '1003')");

        assertUpdate("DELETE FROM " + tableName + " WHERE b = '1001'", 1);
        assertQuery("SELECT * FROM " + tableName, "VALUES (2, '1002'), (3, '1003')");

        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN c integer WITH (partitioning = 'identity')");
        assertUpdate("INSERT INTO " + tableName + " VALUES (1, '1001', 1), (2, '1002', 1), (3, '1003', 2)", 3);
        assertQuery("SELECT * FROM " + tableName, "VALUES (2, '1002', NULL), (3, '1003', NULL), (1, '1001', 1), (2, '1002', 1), (3, '1003', 2)");

        assertUpdate("DELETE FROM " + tableName + " WHERE b = '1002'", 2);
        assertQuery("SELECT * FROM " + tableName, "VALUES (3, '1003', NULL), (1, '1001', 1), (3, '1003', 2)");

        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN d integer WITH (partitioning = 'identity')");
        assertUpdate("INSERT INTO " + tableName + " VALUES (1, '1001', 1, 1), (2, '1002', 1, 2), (3, '1003', 2, 1)", 3);

        assertUpdate("DELETE FROM " + tableName + " WHERE b = '1003'", 3);
        assertQuery("SELECT * FROM " + tableName, "VALUES (1, '1001', 1, NULL), (1, '1001', 1, 1), (2, '1002', 1, 2)");

        dropTable(session, tableName);
    }
```

This test case would fail, the reason might be the partition spec version incompatible mentioned above in `IcebergAbstractMetadata.finishDelete()`.